unified-installer/astra-unified-installer.sh

```diff
@@ -1117,12 +1179,177 @@ is_astra_registry() {
     local -r registry_url="$1"
     if [ -z "$registry_url" ]; then fatal "no registry_url given"; fi
 
-    if echo "$registry" | grep -q "cr." && echo "$registry" | grep -q "astra"; then
+    if echo "$registry_url" | grep -q "cr." && echo "$registry_url" | grep -q "astra"; then
         return 0
     fi
     return 1
 }
 
+# find_astra_registry loops through the given list of registries and returns the first one which matches the
+# pattern of an Astra registry.
+find_astra_registry() {
+    local -ar registries=("$@")
+    for reg in "${registries[@]}" ; do
+        if is_astra_registry "$reg"; then
+            echo "$reg"
+            return 0
+        fi
+    done
+    return 1
+}
+
+find_astra_registry_in_connector_images() {
+    find_astra_registry "$CONNECTOR_IMAGE_REGISTRY" "$NEPTUNE_IMAGE_REGISTRY" "$CONNECTOR_OPERATOR_IMAGE_REGISTRY"
+}
+
+find_astra_registry_in_trident_images() {
+    find_astra_registry "$TRIDENT_OPERATOR_IMAGE_REGISTRY" "$TRIDENT_AUTOSUPPORT_IMAGE_REGISTRY" \
+        "$TRIDENT_IMAGE_REGISTRY" "$TRIDENT_ACP_IMAGE_REGISTRY"
+}
+
+generate_kustomization_dirs_and_files() {
```

Comment:
This used to be a `step_`, but it's been turned into a generic function since we generate three kustomization files now.